nixos/modules/services/search/elasticsearch.nix

```diff
@@ -167,7 +187,8 @@ in
       };
       serviceConfig = {
         ExecStart = "${cfg.package}/bin/elasticsearch ${toString cfg.extraCmdLineOptions}";
-        User = "elasticsearch";
+        User = cfg.user;
+        Group = cfg.group;
         PermissionsStartOnly = true;
```

Comment:
`PermissionsStartOnly` was deprecated years ago... any chance you want to improve this part of the module too?